datafusion/optimizer/src/optimizer.rs

```diff
@@ -375,10 +375,10 @@ impl Optimizer {
 
         let new_inputs = result
             .into_iter()
-            .enumerate()
-            .map(|(i, o)| match o {
+            .zip(inputs)
```

Comment:
One difference with `zip` is that it will silently ignore items if the iterators are different lengths. I reviewed this code and in this case even if results is shorter than inputs (an error) the current code wouldn't error either

Thus I think this is equivalent